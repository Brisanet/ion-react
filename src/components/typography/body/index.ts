export * from './body';
export * from './styles';
