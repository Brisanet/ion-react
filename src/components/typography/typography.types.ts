export type IonTypographyStyle = 'normal' | 'italic';

export type IonTypographyColorScheme =
  | 'primary'
  | 'secondary'
  | 'dark-primary'
  | 'dark-secondary';

interface BaseIonTypography {
  label: string;
  colorScheme?: IonTypographyColorScheme;
}

export type IonTypographySize = 'small' | 'medium' | 'normal';
export type IonTypographyWeight = 'normal' | 'medium' | 'bold';

type IonBodySize<TType> = TType extends 'medium' ? never : TType;
export type IonHeadingWeight<TType> = TType extends 'normal' ? never : TType;

export interface IonBodyInterface extends BaseIonTypography {
  size?: IonBodySize<IonTypographySize>;
  weight?: IonTypographyWeight;
  style?: IonTypographyStyle;
}

export interface IonHeadingInterface extends BaseIonTypography {
  type: 'h1' | 'h2' | 'h3' | 'h4';
  size?: IonTypographySize;
  weight?: IonHeadingWeight<IonTypographyWeight>;
}
