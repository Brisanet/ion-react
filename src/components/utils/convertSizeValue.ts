/**
 * Get a size value based on its type.
 * @param {string | number} value - The size value.
 * @returns The value itself if it's a string, or the value in pixels if it is a number.
 **/
export const convertSizeValue = (value: string | number): string =>
  typeof value === 'number' ? `${value}px` : value;
