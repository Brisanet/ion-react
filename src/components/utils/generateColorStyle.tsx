// Tipo para objeto de estilos  da função geradora de estilos
type ColorStyle = {
    background: string;
    color: string;
    border?: string;
  };
  
  /* 
    Função que gera cores de Background, Border e cor de
    fonte a fim de reduzir repetições de codigo
  
    Seguir o padrão de parâmetros: 
    1- Cor do background;
    2- Cor da fonte do botão;
    3- (Opicional) Cor da borda;
  */
 export function generateColorStyle(
    bgColor: string,
    fontColor: string,
    borderColor: string
  ) {
    let styles: ColorStyle = {
      background: bgColor,
      color: fontColor,
    };
  
    borderColor && (styles.border = `1px solid ${borderColor}`);
  
    return styles;
  }