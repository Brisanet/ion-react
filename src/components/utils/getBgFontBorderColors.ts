interface ObjStyles {
  background: string;
  color: string;
  border?: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param background - Background color
 * @param color - The font color
 * @param border - Border attribute styles - example: ´1px solid blue´
 * @returns A Javascript object containing color properties.
 */
export function getBgFontBorderColors({
  background,
  color,
  border = '',
}: ObjStyles): ObjStyles {
  return {
    background,
    color,
    border,
  };
}
