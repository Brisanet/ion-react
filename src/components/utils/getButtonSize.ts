/**
 * Function to prevent repetition of style blocks.
 *
 * @param buttonSize - Height and min-width
 * @param padding - Button padding
 * @param fontSize - The size of font
 * @returns An Javascript object containing size properties.
 */
export function getButtonSize(
  buttonSize: string,
  padding: string,
  fontSize: string
) {
  let size = {
    height: buttonSize,
    minWidth: buttonSize,
    padding: padding,
    fontSize: fontSize,
  };

  return size;
}
