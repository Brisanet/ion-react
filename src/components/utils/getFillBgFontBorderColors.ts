type ColorStyleResponse = {
  border: string;
  backgroundColor: string;
  color: string;
  svg: {
    fill: string;
  };
};

interface StyleProps {
  svgFillColor: string;
  backgroundColor: string;
  fontColor: string;
  borderColor: string;
}

/**
 * Function to prevent repetition of style blocks.
 *
 * @param svgFillColor - Fill color to be used in SVG
 * @param backgroundColor - Background color
 * @param fontColor - The font color
 * @param borderColor - Border color
 * @returns A Javascript object containing style properties.
 */
export function getFillBgFontBorderColors({
  backgroundColor,
  fontColor,
  borderColor,
  svgFillColor,
}: StyleProps) {
  let styles: ColorStyleResponse = {
    backgroundColor: backgroundColor,
    color: fontColor,
    border: `1px solid ${borderColor}`,
    svg: {
      fill: svgFillColor,
    },
  };

  return styles;
}
