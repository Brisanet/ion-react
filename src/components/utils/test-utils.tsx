import theme from '@ion/styles/theme';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

export const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
