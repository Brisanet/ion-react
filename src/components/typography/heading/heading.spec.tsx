import { renderWithTheme } from '@ion/components/utils/test-utils';
import { screen } from '@testing-library/react';
import { IonHeadingInterface } from '../typography.types';
import IonHeading from './heading';

const defaultProps: IonHeadingInterface = { label: 'Ion React', type: 'h1' };

const sut = (props: IonHeadingInterface = defaultProps) => {
  return renderWithTheme(<IonHeading {...props} />);
};

const headingSize: Array<IonHeadingInterface['size']> = [
  'medium',
  'normal',
  'small',
];
const headingcolorScheme: Array<IonHeadingInterface['colorScheme']> = [
  'dark-primary',
  'dark-secondary',
  'primary',
  'secondary',
];
const headingFontWeight: Array<IonHeadingInterface['weight']> = [
  'bold',
  'medium',
];
const heading: Array<IonHeadingInterface> = [
  { label: 'ion react', type: 'h1' },
  { label: 'ion react', type: 'h2' },
  { label: 'ion react', type: 'h3' },
  { label: 'ion react', type: 'h4' },
];

describe('<IonHeading />', () => {
  it.each(heading)(
    'should render the heading with the correct label $type',
    (heading) => {
      sut(heading);
      expect(
        screen.getByRole('heading', { name: heading.label })
      ).toBeInTheDocument();
    }
  );

  it.each(headingSize)(
    'should render the heading with the correct size %s',
    (size) => {
      const { container } = sut({ ...defaultProps, size });
      expect(container).toMatchSnapshot();
    }
  );

  it.each(headingcolorScheme)(
    'should render the heading with the correct color %s',
    (colorScheme) => {
      const { container } = sut({ ...defaultProps, colorScheme });
      expect(container).toMatchSnapshot();
    }
  );

  it.each(headingFontWeight)(
    'should render the heading with the correct color %s',
    (weight) => {
      const { container } = sut({ ...defaultProps, weight });
      expect(container).toMatchSnapshot();
    }
  );
});
