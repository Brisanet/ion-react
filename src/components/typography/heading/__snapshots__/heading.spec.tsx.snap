// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IonHeading /> should render the heading with the correct color bold 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 600;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct color dark-primary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #fcfcfd;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct color dark-secondary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #e4e6eb;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct color medium 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct color primary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct color secondary 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #505566;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct size medium 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 400;
  font-size: 3.2rem;
  line-height: 4rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct size normal 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 400;
  font-size: 4rem;
  line-height: 4.8rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;

exports[`<IonHeading /> should render the heading with the correct size small 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #282b33;
  font-weight: 400;
  font-size: 2.4rem;
  line-height: 3.2rem;
}

<div>
  <h1
    class="c0"
    data-testid="ion-heading-h1"
  >
    Ion React
  </h1>
</div>
`;
